thrall/app/lib/ThrallMessageConsumer.scala

```diff
@@ -39,21 +39,15 @@ object ThrallMessageConsumer extends MessageConsumer(
   def updateImageUserMetadata(metadata: JsValue): Future[UpdateResponse] =
     withImageId(metadata)(id => ElasticSearch.applyImageMetadataOverride(id, metadata \ "data"))
 
-  def deleteImage(image: JsValue): Future[EsResponse] =
+  // The Unit, Unit is due to the two sied effects
+  def deleteImage(image: JsValue): Future[Either[ImageDeletedResponse, ImageNotDeletableResponse]] =
     withImageId(image) { id =>
-      // if we cannot delete the image as it's "protected", succeed and delete
-      // the message anyway.
-      ElasticSearch.deleteImage(id).map {
-        case r: DeleteByQueryResponse => {
+      ElasticSearch.deleteImage(id).map { deleteResponse =>
+        deleteResponse.left.map { imageDeletedResponse =>
```

Comment:
Not entirely sure what the point of using an `Either` is if we only ever care about the success case?